### PR TITLE
[risk=low][RW-9304] Add Cromwell Pause and Resume buttons to AppsPanel

### PR DESCRIPTION
**Pull request by jmthibault79**
(Update since the video captures: status text improvement, e.g. **Resuming** instead of **STARTING**) 
New Pausing

Pause
Pause

Resume
Resume

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/src/main/resources/leonardo.yaml [hunk 1]
```diff
@@ -1896,11 +1896,16 @@ components:
     AppStatus:
       type: string
       enum:
-        - DELETING
         - DELETED
+        - DELETING
```
Comment (jmthibault79) [inline api/src/main/resources/leonardo.yaml:1900]: These values are present in Leonardo code and returned by endpoints.  They're just missing in the Swagger.  PR to update the Swagger at the source: https://github.com/DataBiosphere/leonardo/pull/3121

File: api/src/main/resources/notebooks.yaml [hunk 2]
```diff
@@ -193,6 +195,94 @@ paths:
...
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorReport"
+        "500":
+          description: Internal Error
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/ErrorReport"
+  "/api/google/v1/apps/{googleProject}/{appName}/start":
```
Comment (jmthibault79) [inline api/src/main/resources/notebooks.yaml:242]: Leonardo "start" = AoU "resume"

File: ui/src/app/components/apps-panel.spec.tsx [hunk 3]
```diff
@@ -86,8 +86,13 @@ describe('AppsPanel', () => {
         findNodesContainingText(wrapper, 'Active applications').exists()
       ).toBe(activeExpected);
 
+      // this changes based on whether there are active applications above this section
+      const availableExpectedHeader = activeExpected
```
Comment (jmthibault79) [inline ui/src/app/components/apps-panel.spec.tsx:90]: not strictly related to this story, but improves the UI a bit

File: ui/src/app/components/apps-panel/expanded-app.tsx [hunk 4]
```diff
@@ -134,14 +150,6 @@ const CromwellButtonRow = (props: {
   );
 };
 
-// TODO: refine and style more like RuntimeStatusIcon
```
Comment (jmthibault79) [inline ui/src/app/components/apps-panel/expanded-app.tsx:137]: we still want this eventually, but for now I moved it inline with the "refresh" text

File: ui/src/app/components/apps-panel/pause-resume-button.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+import { AppsPanelButton } from './apps-panel-button';
+import { UserEnvironmentStatus } from './utils';
+
+interface Props {
+  externalStatus: UserEnvironmentStatus;
+  onPause: Function;
+  onResume: Function;
+}
+export const PauseResumeButton = (props: Props) => {
```
Comment (jmthibault79) [inline ui/src/app/components/apps-panel/pause-resume-button.tsx:16]: RuntimeStateButton converted to `UserEnvironmentStatus` statuses

File: ui/src/app/components/apps-panel/utils.tsx [hunk 6]
```diff
@@ -90,3 +91,31 @@ export const findApp = (
   appType: UIAppType
 ): UserAppEnvironment =>
   apps?.find((app) => app.appType === toAppType(appType));
+
+// used as a generic equivalence for certain states of RuntimeStatus and AppStatus
```
Comment (jmthibault79) [inline ui/src/app/components/apps-panel/utils.tsx:95]: May want to expand this to cover all statuses for RW-9009?

**Comment by jmthibault79** [comment]
e2e failures do not appear to be related

File: api/src/main/resources/leonardo.yaml (hunk 1 above)
Comment (jmthibault79) [reply api/src/main/resources/leonardo.yaml:1900]: Leo Swagger updated.  The main thing here is the addition of the STARTING status.

File: ui/src/app/pages/analysis/leonardo-app-launcher.tsx [hunk 7]
```diff
@@ -16,8 +16,8 @@ import { WithSpinnerOverlayProps } from 'app/components/with-spinner-overlay';
 import { NotebookIcon } from 'app/icons/notebook-icon';
 import { ReminderIcon } from 'app/icons/reminder';
 import {
-  jupyterApi,
-  proxyApi,
+  leoJupyterApi,
```
Comment (jmthibault79) [inline ui/src/app/pages/analysis/leonardo-app-launcher.tsx:19]: renaming only

File: ui/src/app/components/apps-panel/utils.tsx [hunk 8]
```diff
@@ -90,3 +91,38 @@ export const findApp = (
...
+  | 'Resuming';
+
+export const fromRuntimeStatus = (
+  status: RuntimeStatus
+): UserEnvironmentStatus =>
+  cond(
+    [status === RuntimeStatus.Running, () => 'Running'],
+    [status === RuntimeStatus.Stopping, () => 'Pausing'],
+    [status === RuntimeStatus.Stopped, () => 'Paused'],
+    [status === RuntimeStatus.Starting, () => 'resuming'],
```
Comment (jmthibault79) [inline ui/src/app/components/apps-panel/utils.tsx:110]: fixme

**Review by dmohs — APPROVED** [review]
Looks good. Attempted a couple of refactorings, but liked the original better in both cases.

**Comment by jmthibault79** [comment]
api-integration-test failure is related to ToS, not the changes in this branch (main is failing too)

**Review by evrii — APPROVED** [review]
Looks good to me.

**Pull request merged**